src/StokesDrifts.jl

```diff
@@ -0,0 +1,225 @@
+
+stokes_drift_parameters = (uˢ₀ = 0.005, h = 20)
+stokes_drift = UniformStokes(∂z_uˢ=uniform_stokes_shear, parameters=stokes_drift_parameters)
+```
+
+"""
+UniformStokesDrift(; ∂z_uˢ=addzero, ∂z_vˢ=addzero, ∂t_uˢ=addzero, ∂t_vˢ=addzero, parameters=nothing) =
+    UniformStokesDrift(∂z_uˢ, ∂z_vˢ, ∂t_uˢ, ∂t_vˢ, parameters)
+
+const USD = UniformStokesDrift
+const USDnoP = UniformStokesDrift{<:Nothing}
+const f = Face()
+const c = Center()
+
+@inline ∂t_uˢ(i, j, k, grid, sw::USD, time) = sw.∂t_uˢ(znode(k, grid, c), time, sw.parameters)
+@inline ∂t_vˢ(i, j, k, grid, sw::USD, time) = sw.∂t_vˢ(znode(k, grid, c), time, sw.parameters)
+@inline ∂t_wˢ(i, j, k, grid, sw::USD, time) = zero(grid)
+
+@inline x_curl_Uˢ_cross_U(i, j, k, grid, sw::USD, U, time) = @inbounds    ℑxzᶠᵃᶜ(i, j, k, grid, U.w) * sw.∂z_uˢ(znode(k, grid, c), time, sw.parameters)
+@inline y_curl_Uˢ_cross_U(i, j, k, grid, sw::USD, U, time) = @inbounds    ℑyzᵃᶠᶜ(i, j, k, grid, U.w) * sw.∂z_vˢ(znode(k, grid, c), time, sw.parameters)
+
+@inline z_curl_Uˢ_cross_U(i, j, k, grid, sw::USD, U, time) = @inbounds (- ℑxzᶜᵃᶠ(i, j, k, grid, U.u) * sw.∂z_uˢ(znode(k, grid, f), time, sw.parameters)
+                                                                        - ℑyzᵃᶜᶠ(i, j, k, grid, U.v) * sw.∂z_vˢ(znode(k, grid, f), time, sw.parameters) )
+
+# Methods for when `parameters == nothing`
+@inline ∂t_uˢ(i, j, k, grid, sw::USDnoP, time) = sw.∂t_uˢ(znode(k, grid, c), time)
+@inline ∂t_vˢ(i, j, k, grid, sw::USDnoP, time) = sw.∂t_vˢ(znode(k, grid, c), time)
+
+@inline x_curl_Uˢ_cross_U(i, j, k, grid, sw::USDnoP, U, time) = @inbounds    ℑxzᶠᵃᶜ(i, j, k, grid, U.w) * sw.∂z_uˢ(znode(k, grid, c), time)
+@inline y_curl_Uˢ_cross_U(i, j, k, grid, sw::USDnoP, U, time) = @inbounds    ℑyzᵃᶠᶜ(i, j, k, grid, U.w) * sw.∂z_vˢ(znode(k, grid, c), time)
+@inline z_curl_Uˢ_cross_U(i, j, k, grid, sw::USDnoP, U, time) = @inbounds (- ℑxzᶜᵃᶠ(i, j, k, grid, U.u) * sw.∂z_uˢ(znode(k, grid, f), time)
+                                                                           - ℑyzᵃᶜᶠ(i, j, k, grid, U.v) * sw.∂z_vˢ(znode(k, grid, f), time))
+
+struct StokesDrift{P, VX, WX, UY, WY, UZ, VZ, UT, VT, WT}
+    ∂x_vˢ :: VX
+    ∂x_wˢ :: WX
+    ∂y_uˢ :: UY
+    ∂y_wˢ :: WY
+    ∂z_uˢ :: UZ
+    ∂z_vˢ :: VZ
+    ∂t_uˢ :: UT
+    ∂t_vˢ :: VT
+    ∂t_wˢ :: WT
+    parameters :: P
+end
+
+"""
+    StokesDrift(; ∂z_uˢ=addzero, ∂y_uˢ=addzero, ∂t_uˢ=addzero, 
+                  ∂z_vˢ=addzero, ∂x_vˢ=addzero, ∂t_vˢ=addzero, 
+                  ∂x_wˢ=addzero, ∂y_wˢ=addzero, ∂t_wˢ=addzero, parameters=nothing)
+
+Construct a set of functions of space and time for a Stokes drift velocity field
+corresponding to a surface gravity wave field with an envelope that (potentially) varies
+in the horizontal directions.
+
+To resolve the evolution of the Lagrangian-mean momentum, we require all the components
+of the "psuedovorticity",
+
+```math
+∇ × uˢ = x̂ (∂y_wˢ - ∂z_vˢ) + ŷ (∂z_uˢ - ∂x_wˢ) + ẑ (∂x_vˢ - ∂y_uˢ)
```

Comment:
```suggestion
∇ × uˢ = x̂ (∂_y wˢ - ∂_z vˢ) + ŷ (∂_z uˢ - ∂_x wˢ) + ẑ (∂_x vˢ - ∂_y uˢ)
```

Reply:
this is math so i will accept the change